tests/postUI.test.tsx

```diff
@@ -0,0 +1,145 @@
+  name: "Post Test",
+  isVerified: true
+}
+
+describe("PostUI Component Tests", () => {
+  const props = {
+    id: '5',
+    createdAt: "2023-01-23T06:35:53.184Z",
+    updatedAt: "2023-01-29T01:42:01.713Z",
+    userId: "078ff5c0-5bce-4603-b1f3-79cf8258ec26",
+    description: "mockDescription",
+    channel: "general"
+  }
+  const post: Post = new Post(props);
+  const mockReportPost = jest.fn();
+  const mockOps = {
+    removeItem: jest.fn(),
+    replaceItem: jest.fn()
+  }
+  const time = new Date();
+  
+  it("Renders with all options", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={time}
+      maxOccupancy={8} hasInvitations={true}
+    />);
+    
+    expect(screen.getByText(props.description)).toBeDefined();
+    expect(screen.getByLabelText('time until')).toBeDefined();
+    expect(screen.getByLabelText('max occupancy')).toBeDefined();
+    expect(screen.getByLabelText('request invitations')).toBeDefined();
+  });
+
+  it("Renders with only time until event starts", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={time}
+      maxOccupancy={undefined} hasInvitations={false}
+    />);
+    
+    expect(screen.getByText(props.description)).toBeDefined();
+    expect(screen.queryByLabelText('time until')).toBeDefined();
+    expect(screen.queryByLabelText('max occupancy')).toBeNull();
+    expect(screen.queryByLabelText('request invitations')).toBeNull();
+  });
+
+  it("Renders with only max occupancy", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={undefined}
+      maxOccupancy={8} hasInvitations={false}
+    />);
+    
+    expect(screen.getByText(props.description)).toBeDefined();
+    expect(screen.queryByLabelText('time until')).toBeNull();
+    expect(screen.queryByLabelText('max occupancy')).toBeDefined();
+    expect(screen.queryByLabelText('request invitations')).toBeNull();
+  });
+
+  it("Renders with only invitations", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={undefined}
+      maxOccupancy={undefined} hasInvitations={true}
+    />);
+    
+    expect(screen.getByText(props.description)).toBeDefined();
+    expect(screen.queryByLabelText('time until')).toBeNull();
+    expect(screen.queryByLabelText('max occupancy')).toBeNull();
+    expect(screen.queryByLabelText('request invitations')).toBeDefined();
+  });
+
+  it("Name is shortened", () => {
+    render(<PostItem item={post} likes={0} reportPost={mockReportPost}
+      writtenByYou={false} operations={mockOps} startTime={time}
+      maxOccupancy={8} hasInvitations={true}
+    />);
+    
+    expect(screen.getByText("Post T.")).toBeDefined();
```

Comment:
Is it possible to check if the name is truncated? If so, could we also have a test that checks if a long description is truncated?